project_functional_block/models/project_functional_block.py

```diff
@@ -0,0 +1,146 @@
+
+from openerp import models, fields, api
+
+
+class ProjectFunctionalBlock(models.Model):
+    _name = 'project.functional.block'
+    _description = 'Functional block to organize projects tasks'
+    _order = 'parent_id desc, name'
+    _rec_name = 'complete_name'
+    _parent_store = True
+    _parent_order = 'name'
+
+    # Columns
+    name = fields.Char('Name', size=64, required=True, translate=True)
+    code = fields.Char('Code', size=8)
+    complete_name = fields.Char(
+        string='Complete Name', compute='_compute_complete_name')
+    description = fields.Text('Description', translate=True)
+    parent_id = fields.Many2one(
+        'project.functional.block', 'Parent Block',
+        ondelete='cascade', index=True)
+    parent_left = fields.Integer('Left Parent', index=True)
+    parent_right = fields.Integer('Right Parent', index=True)
+    tasks_count = fields.Integer(string='Tasks Count', compute='_count_tasks')
+    child_fblocks_count = fields.Integer(
+        string='Child Blocks Count', compute='_count_child_fblocks')
+
+    @api.multi
+    def _compute_complete_name(self):
+        """
+        Complete Name includes parent's name and name of this functional block
+        itself.
+        """
+        todo_fblocks = self.search([('id', 'child_of', self.ids)],
+                                   order='parent_id desc, name')
+        for func_block in todo_fblocks:
+            complete_name = func_block.name
+            parent = func_block.parent_id
+            while parent:
+                complete_name = parent.name + ' / ' + complete_name
+                parent = parent.parent_id
+            func_block.complete_name = complete_name
+
+    @api.multi
+    def _count_tasks(self):
+        """
+        Count the number of tasks of given functional block(s).
+        """
+        task_obj = self.env['project.task']
+        for fblock in self:
+            tasks_count = task_obj.search(
+                [('functional_block_id', '=', fblock.id)], count=True)
+            fblock.tasks_count = tasks_count
+
+    @api.multi
+    def action_view_tasks(self):
+        """
+        View tasks of the given block.
+        """
+        self.ensure_one()
+        action = self.env.ref('project.action_view_task')
+        task_obj = self.env['project.task']
+        tasks = task_obj.search([('functional_block_id', '=', self.id)])
+        task_ids_str = ','.join(map(str, tasks.ids))
+        ret_ctx = {
+            'default_functional_block_id': self.id
+        }
+        return {
+            'name': action.name,
+            'help': action.help,
+            'type': action.type,
+            'view_type': action.view_type,
+            'view_mode': action.view_mode,
+            'target': action.target,
+            'context': str(ret_ctx),
+            'res_model': action.res_model,
+            'domain': "[('id', 'in', (" + task_ids_str + ",))]"
+        }
+
+    @api.multi
+    def _count_child_fblocks(self):
```

Comment:
If you add a `child_ids` you can then:

    for fblock in self:
        fblock.tasks_count = sum(x.tasks_count fro x in fblock.child_ids)